Sources/MongoSwift/MongoCursor.swift

```diff
@@ -187,47 +182,60 @@ public class MongoCursor<T: Codable>: Sequence, IteratorProtocol {
     }
 
     /**
-     * Returns the next `Document` in this cursor or `nil`, or throws an error if one occurs -- compared to `next()`,
-     * which returns `nil` and requires manually checking for an error afterward.
-     * - Returns: the next `Document` in this cursor, or `nil` if at the end of the cursor
+     * Returns an array of type  `T` from this cursor, or throws
```

Comment:
nit: the fact that this uses `Result`s and gets errors from them is an implementation detail that doesn't really matter for the user. I think you could probably just omit everything after "this cursor" .